src/monitor-shim.ts

```diff
@@ -43,8 +43,15 @@ export default class ShimMonitor extends Monitor {
       enumerable: this.enumerable,
       get: () => this.state,
       set: (value: any) => {
+        // checks if the incoming value matches the existing value
+        const hasChanged = this.state !== value;
```

Comment:
Just for my own kownledge: `this.state` is the current value of the property, and we are doing a shallow compare here to only emit the change event when they are not the same, right? 
When the property is an object, do we recursively shim, or is that even a valid configuration?

Reply:
Yes.  This is really to guard against primitive assignments that are the same.  For objects, we don't do a deep comparison to figure out if the content itself is different.  It feels like the performance cost of doing that probably outweighs any downstream benefit.

DLO does recursively shim. There's some technical nuance that probably needs revisiting, but if the target is `digitalData.cart`, we shim all properties in `cart` and we _should_ be shimming all properties in child objects.  This monitor code looks at the property specifically for changes, and if it happens it's emitted.  So if you did:

```
digitalData.cart.cartID = '1234';
digitalData.cart.total = 29.99;
```

That would be two distinct changes that we'd rely on debouncing to help smooth sequential changes into a single "event".